toolchain/newlib-4.3.0-oak.patch

```diff
@@ -0,0 +1,508 @@
+ 	-rm -f Makefile
+ distclean-am: clean-am distclean-compile distclean-generic \
+ 	distclean-hdr distclean-local distclean-tags
+@@ -4382,7 +4443,7 @@
+ maintainer-clean: maintainer-clean-recursive
+ 	-rm -f $(am__CONFIG_DISTCLEAN_FILES)
+ 	-rm -rf $(top_srcdir)/autom4te.cache
+-	-rm -rf ./$(DEPDIR) aarch64/$(DEPDIR) arc/$(DEPDIR) arm/$(DEPDIR) bfin/$(DEPDIR) csky/$(DEPDIR) d30v/$(DEPDIR) iq2000/$(DEPDIR) libnosys/$(DEPDIR) lm32/$(DEPDIR) nios2/$(DEPDIR) riscv/$(DEPDIR)
++	-rm -rf ./$(DEPDIR) aarch64/$(DEPDIR) arc/$(DEPDIR) arm/$(DEPDIR) bfin/$(DEPDIR) csky/$(DEPDIR) d30v/$(DEPDIR) iq2000/$(DEPDIR) libnosys/$(DEPDIR) lm32/$(DEPDIR) nios2/$(DEPDIR) oak/$(DEPDIR) riscv/$(DEPDIR)
+ 	-rm -f Makefile
+ maintainer-clean-am: distclean-am maintainer-clean-aminfo \
+ 	maintainer-clean-generic maintainer-clean-local
+diff '--color=auto' -r -N -u newlib-4.3.0.20230120.orig/libgloss/oak/crt0.S newlib-4.3.0.20230120/libgloss/oak/crt0.S
+--- newlib-4.3.0.20230120.orig/libgloss/oak/crt0.S	1970-01-01 00:00:00.000000000 +0000
++++ newlib-4.3.0.20230120/libgloss/oak/crt0.S	2023-03-03 17:51:31.496517802 +0000
+@@ -0,0 +1,38 @@
++#
++# Copyright 2023 The Project Oak Authors
++#
++# Licensed under the Apache License, Version 2.0 (the "License");
++# you may not use this file except in compliance with the License.
++# You may obtain a copy of the License at
++#
++#     http://www.apache.org/licenses/LICENSE-2.0
++#
++# Unless required by applicable law or agreed to in writing, software
++# distributed under the License is distributed on an "AS IS" BASIS,
++# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
++# See the License for the specific language governing permissions and
++# limitations under the License.
++#
++
++.text
++.global _start
++
++_start:
++    movq $0, %rbp
++    pushq %rbp
++    pushq %rbp
++    movq %rsp, %rbp
++
++    call _init
++
++    # argc = 0, argv = NULL, envp = NULL
++    xor %rdi, %rdi
++    xor %rsi, %rsi
++    xor %rdx, %rdx
++    call main
++    mov %rax, %rdi # rdi = rax
++    call _exit
++
++3:
++    nop
++    jmp 3b
+diff '--color=auto' -r -N -u newlib-4.3.0.20230120.orig/libgloss/oak/Makefile.inc newlib-4.3.0.20230120/libgloss/oak/Makefile.inc
+--- newlib-4.3.0.20230120.orig/libgloss/oak/Makefile.inc	1970-01-01 00:00:00.000000000 +0000
++++ newlib-4.3.0.20230120/libgloss/oak/Makefile.inc	2023-03-03 17:51:48.669958515 +0000
+@@ -0,0 +1,5 @@
++multilibtool_LIBRARIES += %D%/libgloss.a
++%C%_libgloss_a_CPPFLAGS = -I$(srcdir)/%D%
++%C%_libgloss_a_SOURCES = \
++    %D%/syscalls.c \
++    %D%/crt0.S
+diff '--color=auto' -r -N -u newlib-4.3.0.20230120.orig/libgloss/oak/syscalls.c newlib-4.3.0.20230120/libgloss/oak/syscalls.c
+--- newlib-4.3.0.20230120.orig/libgloss/oak/syscalls.c	1970-01-01 00:00:00.000000000 +0000
++++ newlib-4.3.0.20230120/libgloss/oak/syscalls.c	2023-03-03 17:55:18.491557542 +0000
```

Comment:
IIUC this is the main Oak-specific syscall interface -- should we actually check it in as a separate file, and copy that in in addition to the patch? so it's easier to maintain (and it could do with more documentation, e.g. around syscall numbers)

Reply:
Yep, that's probably a good idea. I'll submit the chain as-is, but will come back to it later.